examples/jsm/tsl/shadows/TileShadowNode.js

```diff
@@ -0,0 +1,452 @@
+import {
+	Vector3,
+	Object3D,
+	ShadowBaseNode,
+	Plane,
+	Line3,
+	DepthArrayTexture,
+	LessCompare,
+	Vector2,
+	RedFormat,
+	NodeMaterial,
+	ArrayCamera,
+	VSMShadowMap,
+	NodeUtils,
+	RendererUtils,
+	Quaternion } from 'three/webgpu';
+
+import { min, Fn, shadow, vec4, positionWorld, renderGroup, reference, objectPosition } from 'three/tsl';
+import { TileShadowNodeHelper } from './TileShadowNodeHelper.js';
+
+const { getDataFromObject } = NodeUtils;
+const { resetRendererAndSceneState, restoreRendererAndSceneState } = RendererUtils;
+let _rendererState;
+
+const shadowMaterialLib = /*@__PURE__*/ new WeakMap();
+const linearDistance = /*@__PURE__*/ Fn( ( [ position, cameraNear, cameraFar ] ) => {
+
+	let dist = positionWorld.sub( position ).length();
+	dist = dist.sub( cameraNear ).div( cameraFar.sub( cameraNear ) );
+	dist = dist.saturate(); // clamp to [ 0, 1 ]
+
+	return dist;
+
+} );
+
+const linearShadowDistance = ( light ) => {
+
+	const camera = light.shadow.camera;
+
+	const nearDistance = reference( 'near', 'float', camera ).setGroup( renderGroup );
+	const farDistance = reference( 'far', 'float', camera ).setGroup( renderGroup );
+
+	const referencePosition = objectPosition( light );
+
+	return linearDistance( referencePosition, nearDistance, farDistance );
+
+};
+
+const getShadowMaterial = ( light ) => {
+
+	let material = shadowMaterialLib.get( light );
+
+	if ( material === undefined ) {
+
+		const depthNode = light.isPointLight ? linearShadowDistance( light ) : null;
+
+		material = new NodeMaterial();
+		material.colorNode = vec4( 0, 0, 0, 1 );
+		material.depthNode = depthNode;
+		material.isShadowPassMaterial = true; // Use to avoid other overrideMaterial override material.colorNode unintentionally when using material.shadowNode
+		material.name = 'ShadowMaterial';
+		material.fog = false;
+
+		shadowMaterialLib.set( light, material );
+
+	}
+
+	return material;
+
+};
+
+
+class LwLight extends Object3D {
+
+	constructor() {
+
+		super();
+		this.target = new Object3D();
+
+	}
+
+}
+class TileShadowNode extends ShadowBaseNode {
+
+	constructor( light, camera, scene, options = {} ) {
```

Comment:
Can we remove the `camera` and `scene` parameters and get them through `builder.camera` and `builder.scene`? I think they should be the same as in the current pass to match with the projection calculations.